src/middlewared/middlewared/plugins/truecommand/wireguard.py

```diff
@@ -0,0 +1,90 @@
+import re
+import subprocess
+import time
+
+from middlewared.service import CallError, periodic, private, Service
+from middlewared.utils import Popen, run
+
+from .enums import Status
+
+WIREGUARD_HEALTH_RE = re.compile(r'=\s*(.*)')
+
+
+class TruecommandService(Service):
+
+    HEALTH_CHECK_SECONDS = 1800
+
+    @private
+    async def generate_wg_keys(self):
+        cp = await Popen(['wg', 'genkey'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        private_key, stderr = await cp.communicate()
+        if cp.returncode:
+            raise CallError(
+                f'Failed to generate key for wireguard with exit code ({cp.returncode}): {stderr.decode()}'
+            )
+
+        cp = await Popen(
+            ['wg', 'pubkey'], stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        public_key, stderr = await cp.communicate(input=private_key)
+        if cp.returncode:
+            raise CallError(
+                f'Failed to generate public key for wireguard with exit code ({cp.returncode}): {stderr.decode()}'
+            )
+
+        return {'wg_public_key': public_key.decode().strip(), 'wg_private_key': private_key.decode().strip()}
+
+    @private
+    @periodic(1800, run_on_start=False)
```

Comment:
HEALTH_CHECK_SECONDS instead of 1800?

Reply:
@sonicaj this was resolved but its still 1800 and no explanation why ;)

Reply:
Because I have fixed it locally and hadn't pushed, waiting for your reply on the first comment ;) ( i usually mark comments as resolved when I have fixed them locally so as to know which comments are left to take care of, so no worries, pushing asap )